### PR TITLE
Update HDF5 I/O to use new HDF5 Async VOL APIs

### PR DESCRIPTION
**Pull request by houjun**
## Summary
Update the HDF5 I/O to (optionally) use the latest HDF5 Async VOL EventSet APIs (*_async) when enabled in GNUMakefile, the original HDF5 I/O is not affected if Async VOL is not enabled.

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

File: Src/Base/AMReX_PlotFileUtil.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 #ifdef AMREX_USE_HDF5_ASYNC
 #include "h5_vol_external_async_native.h"
+hid_t es_id_g = 0;
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_PlotFileUtil.cpp]: Can we put this inside `namespace {}`?

File: Src/Particle/AMReX_Particles.H [hunk 2]
```diff
@@ -48,6 +48,10 @@
 
 #ifdef AMREX_USE_HDF5
 #include <hdf5.h>
+#ifdef AMREX_USE_HDF5_ASYNC
+#include "h5_vol_external_async_native.h"
+extern hid_t es_par_g;
```
Comment (WeiqunZhang) [inline Src/Particle/AMReX_Particles.H]: Could we prefix this global variable with `amrex_` or put it in `namespace amrex`?

Comment (houjun) [reply Src/Particle/AMReX_Particles.H]: @WeiqunZhang  Just moved them (es_par_g and es_id_g) to amrex namespace.

**Comment by houjun** [comment]
Hi @atmyers, got a chance to review this?

**Review by atmyers — APPROVED** [review]

**Pull request merged**